lib/ecto/repo/supervisor.ex

```diff
@@ -154,6 +154,29 @@ defmodule Ecto.Repo.Supervisor do
     end
   end
 
+  @compile {:inline, triplet: 2, maybe_put_stacktrace: 2}
+
+  @doc false
+  def triplet(name, opts) do
+    {adapter, adapter_meta} = Ecto.Repo.Registry.lookup(name)
+      
+    {adapter, adapter_meta, maybe_put_stacktrace(opts, adapter_meta)}
+  end
+
+  defp maybe_put_stacktrace(opts, adapter_meta) do
+    if opts[:get_stacktrace?] || adapter_meta[:get_stacktrace?] do
```

Comment:
Hrm, I am not sure this is the best name for the option. It makes sense from internally, but from the user perspective, they are not getting any stacktrace. I would go with `:stacktrace` or `:include_stacktrace_in_telemetry` or similar.

```suggestion
    if opts[:get_stacktrace?] || adapter_meta[:get_stacktrace?] do
```

Reply:
And then we need to document it. :)

Reply:
I'll go with stacktrace, because it may also be used in logs, so i think it is a better option